examples/diesel/src/main.rs

```diff
@@ -43,46 +44,50 @@ struct RowsUpdated {
     rows: usize,
 }
 
-fn create_product_handler(mut state: State) -> Box<HandlerFuture> {
+fn create_product_handler(mut state: State) -> Pin<Box<HandlerFuture>> {
     let repo = Repo::borrow_from(&state).clone();
-    let f = extract_json::<NewProduct>(&mut state)
-        .and_then(move |product| {
-            repo.run(move |conn| {
-                // Insert the `NewProduct` in the DB
+    async move {
+        let product = match extract_json::<NewProduct>(&mut state).await {
+            Ok(product) => product,
+            Err(e) => return Err((state, e)),
+        };
+
+        let rows = match repo
+            .run(move |conn| {
                 diesel::insert_into(products::table)
                     .values(&product)
                     .execute(&conn)
             })
-            .map_err(|e| e.into_handler_error())
-        })
-        .then(|result| match result {
-            Ok(rows) => {
-                let body = serde_json::to_string(&RowsUpdated { rows })
-                    .expect("Failed to serialise to json");
-                let res =
-                    create_response(&state, StatusCode::CREATED, mime::APPLICATION_JSON, body);
-                future::ok((state, res))
-            }
-            Err(e) => future::err((state, e)),
-        });
-    Box::new(f)
+            .await
+        {
+            Ok(rows) => rows,
+            Err(e) => return Err((state, e.into_handler_error())),
+        };
+
+        let body =
+            serde_json::to_string(&RowsUpdated { rows }).expect("Failed to serialise to json");
+        let res = create_response(&state, StatusCode::CREATED, mime::APPLICATION_JSON, body);
+        Ok((state, res))
+    }
+    .boxed()
```

Comment:
I haven't checked recently, but wasn't this form basically deprecated in favour of `Box::new`?

Reply:
Rather than repeat, this comment applies to all `.boxed()`.

Reply:
[`.boxed()`][0] is part of the `FutureExt` trait extension.  It's just a helper for `Box::pin`.  In fact, that's the [entire implementation][1].

[0]: https://docs.rs/futures/0.3.1/futures/future/trait.FutureExt.html#method.boxed
[1]: https://github.com/rust-lang/futures-rs/blob/f6ee3bdf925af52b67c4258618dad7e011c8f648/futures-util/src/future/future/mod.rs#L463